packages/cli/generators/app/templates/src/application.ts.ejs

```diff
@@ -19,6 +19,17 @@ export class <%= project.applicationName %> extends BootMixin(
 export class <%= project.applicationName %> extends BootMixin(RestApplication) {
 <% } -%>
   constructor(options: ApplicationConfig = {}) {
+    options = Object.assign(
```

Comment:
Could you please verify that `Object.assign` is preserving deep values provided by `options`? I believe it will replace `rest` object as a whole, i.e. changing `{rest: {port: 123, host: 'name'}}` provided by the top-level `index.js` into `{rest: {openApiSpec: {setServersFromRequest: true}}}`. A possible solution is to use a deep-merge algorithm, there are many implementations available on npmjs. [deepmerge](https://www.npmjs.com/package/deepmerge) seems like a popular one that's still actively maintained. `lodash` provides deep-merge or deep-assign API too.

Reply:
Why not move the config to `index.js.ejs`?

Reply:
That could also be a good solution @raymondfeng , simplifies everything for this context. @bajtos  thoughts?